test/unit/deferred.js

```diff
@@ -759,10 +756,8 @@ QUnit.test( "jQuery.when - joined", function( assert ) {
 				expected = shouldResolve ? [ 1, 1 ] : [ 0, undefined ],
 				expectedNotify = shouldNotify && [ willNotify[ id1 ], willNotify[ id2 ] ],
 				code = "jQuery.when( " + id1 + ", " + id2 + " )",
-				context1 = defer1 && jQuery.isFunction( defer1.promise ) ? defer1.promise() :
-					( defer1.then ? window : undefined ),
-				context2 = defer2 && jQuery.isFunction( defer2.promise ) ? defer2.promise() :
-					( defer2.then ? window : undefined );
+				context1 = defer1 && jQuery.isFunction( defer1.promise ) ? defer1.promise() : window,
+				context2 = defer2 && jQuery.isFunction( defer2.promise ) ? defer2.promise() : window;
```

Comment:
But, why? What's the point of returning `undefined` instead of `window`, or vice versa? I imagine neither is being used very often. And why are we continuing to do different things for `$.when( 1 )` and `$.when( 1, 1 )`. jQuery 1.8.3 looks just as bad, only swapped. Time to get them to do the same thing.